### PR TITLE
Work/add logging to etagwatcher and scheduler

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

**Comment by erikjv** [comment]
I'm not sure what to log in the `SyncScheduler`, some pointers would be useful.

File: src/gui/scheduling/etagwatcher.cpp [hunk 1]
```diff
@@ -81,6 +83,7 @@ void ETagWatcher::updateEtag(Folder *f, const QString &etag)
     if (OC_ENSURE_NOT(etag.isEmpty())) {
         auto &info = _lastEtagJob[f];
         if (f->canSync() && info.etag != etag) {
+            qCDebug(lcEtagWatcher) << "Scheduling etag update for space" << f->displayName();
```
Comment (TheOneRing) [inline src/gui/scheduling/etagwatcher.cpp:86]: qCDebug(lcEtagWatcher) << "Scheduling sync of" << f->displayName() << "due to an etag change";

**Comment by TheOneRing** [comment]
Please also add logs to the scheduler if you think it can make tracing based on logs easier.

File: src/gui/scheduling/etagwatcher.cpp (hunk 1 above)
Comment (erikjv) [reply src/gui/scheduling/etagwatcher.cpp]: done

**Review by TheOneRing — CHANGES_REQUESTED** [review]
nitpick ^

File: src/gui/scheduling/syncscheduler.cpp [hunk 2]
```diff
@@ -143,10 +151,13 @@ void SyncScheduler::startNext()
         connect(
             _currentSync, &Folder::syncFinished, this,
             [this](const SyncResult &result) {
+                qCInfo(lcSyncScheduler) << "Sync finished for" << _currentSync->displayName() << "with status" << Utility::enumToDisplayName(result.status());
```
[the remark this answers is not in the document]
Comment (erikjv) [reply src/gui/scheduling/syncscheduler.cpp]: done

File: src/gui/scheduling/syncscheduler.cpp [hunk 3]
```diff
@@ -143,10 +151,13 @@ void SyncScheduler::startNext()
         connect(
             _currentSync, &Folder::syncFinished, this,
             [this](const SyncResult &result) {
+                qCInfo(lcSyncScheduler) << "Sync finished for" << _currentSync->displayName() << "with status" << result.status();
                 if (result.status() != SyncResult::Success) {
                     // Retry a couple of times after failure; or regularly if requested
-                    if (_currentSync->canSync() && (_currentSync->consecutiveFailingSyncs() > 0 && _currentSync->consecutiveFailingSyncs() < 3)
+                    if ((_currentSync->canSync() && (_currentSync->consecutiveFailingSyncs() > 0 && _currentSync->consecutiveFailingSyncs() < 3))
                         || _currentSync->syncEngine().isAnotherSyncNeeded() == AnotherSyncNeeded::DelayedFollowUp) {
+                        qCInfo(lcSyncScheduler) << "Rescheduling sync for" << _currentSync->displayName() << "in" << SyncEngine::minimumFileAgeForUpload
```
Comment (TheOneRing) [inline src/gui/scheduling/syncscheduler.cpp:159]: We have a pretty printer for std::chrono.
`23-09-22 10:07:27:153 [ info scheduler.syncscheduler ]:	Rescheduling sync for "Personal" in duration(0h, 0min, 2s, 0ms) seconds`
As the delay is well, not very important we can just omit it from the log?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/gui/scheduling/syncscheduler.cpp (hunk 3 above)
Comment (erikjv) [reply src/gui/scheduling/syncscheduler.cpp:159]: will do, wasn't sure about that.

File: src/gui/scheduling/syncscheduler.cpp [hunk 4]
```diff
@@ -143,18 +151,31 @@ void SyncScheduler::startNext()
...
+
+                    // Retry a couple of times after failure; or regularly if requested
+                    if (_currentSync->canSync() && _currentSync->consecutiveFailingSyncs() > 0) {
+                        if (_currentSync->consecutiveFailingSyncs() < 3) {
+                            qCInfo(lcSyncScheduler) << "Sync failed, rescheduling sync for" << _currentSync->displayName();
+                            reschedule();
+                        } else {
+                            qCInfo(lcSyncScheduler) << "Sync for" << _currentSync->displayName() << "failed 3 times, not rescheduling";
+                        }
+                    } else if (_currentSync->syncEngine().isAnotherSyncNeeded() == AnotherSyncNeeded::DelayedFollowUp) {
```
Comment (TheOneRing) [inline src/gui/scheduling/syncscheduler.cpp:168]: Hmm now you would reschedule evewn if we can't sync?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

**Pull request merged**